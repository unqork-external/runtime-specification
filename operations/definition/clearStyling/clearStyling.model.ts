import { Description, DiscriminatorValue, Required } from '@tsed/schema'

import { ClearStylingOperationOptions } from './clearStyling.options'
import { OperationTypes } from '../../enums/operation-types.enum'
import { Operation } from '../../interface/operations.interface'

@DiscriminatorValue(OperationTypes.CLEAR_STYLING)
@Description('This operation will clear any styles added to a component via `styling` configuration.')
export class ClearStylingOperation extends Operation<OperationTypes.CLEAR_STYLING> {
  @Required()
  public options: ClearStylingOperationOptions = new ClearStylingOperationOptions()
}
