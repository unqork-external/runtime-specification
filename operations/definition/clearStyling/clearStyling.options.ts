import { TargetedOperationOptions } from '../../interface'

export class ClearStylingOperationOptions extends TargetedOperationOptions {}
