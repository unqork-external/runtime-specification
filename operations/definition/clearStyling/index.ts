export * from './clearStyling.model'
export * from './clearStyling.options'
