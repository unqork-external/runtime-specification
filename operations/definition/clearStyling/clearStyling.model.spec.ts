import { ClearStylingOperation } from './clearStyling.model'
import { generateSchemaAndValidate } from '../../../../utilities'

describe('Clear Styling Operation Model', () => {
  let schema
  beforeAll(() => {
    const response = generateSchemaAndValidate(ClearStylingOperation)
    schema = response.schema
  })

  it('should match this JSON Schema', () => {
    expect(schema).toMatchSnapshot()
  })
})
