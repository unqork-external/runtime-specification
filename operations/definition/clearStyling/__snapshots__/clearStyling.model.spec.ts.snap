// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Clear Styling Operation Model should match this JSON Schema 1`] = `
{
  "autogenDisabled": true,
  "definitions": {
    "ClearStylingOperationOptions": {
      "properties": {
        "shouldExecute": {
          "default": true,
          "description": "
    Conditionally execute this operation.
    Can set to false to temporarily disabled when debugging.
  ",
          "type": [
            "string",
            "boolean",
          ],
        },
        "targetKey": {
          "description": "Target component(s) that will receive the \`styling\` update. Engine Syntax is supported as a way to access the intended targets.",
          "examples": [
            "styled-component",
            "panel.*",
          ],
          "type": "string",
        },
      },
      "type": "object",
    },
  },
  "description": "This operation will clear any styles added to a component via \`styling\` configuration.",
  "properties": {
    "creatorSummary": {
      "description": "A detailed summary of the operation",
      "type": "string",
    },
    "name": {
      "description": "Name of the operation",
      "type": "string",
    },
    "options": {
      "$ref": "#/definitions/ClearStylingOperationOptions",
    },
    "type": {
      "const": "CLEAR_STYLING",
      "examples": [
        "CLEAR_STYLING",
      ],
      "type": "string",
    },
  },
  "required": [
    "options",
  ],
  "type": "object",
}
`;
